viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not bool(title) or not bool(genre) or not bool(rating):
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    if not bool(movie):
+        return user_data
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    if not bool(movie):
+        return user_data
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:

```

Comment:
This is working here, but an important note is that it is generally dangerous practice to modify a data structure that you are using to loop over.  This can lead to bugs in more complex code.  One alternative could be making a copy of what you get from " movie in user_data["watchlist"]:" so that that collection does not change while you modify user_data["watchlist]" from inside the loop.